Makefile.uk

```diff
@@ -0,0 +1,121 @@
+
+################################################################################
+# Helpers
+################################################################################
+LIBPNG_EXTRACTED = $(LIBPNG_ORIGIN)/libpng-$(LIBPNG_VERSION)
+
+################################################################################
+# Library includes
+################################################################################
+LIBPNG_COMMON_INCLUDES-y      += -I$(LIBPNG_EXTRACTED)/
+
+CINCLUDES-$(CONFIG_LIBPNG)    += $(LIBPNG_COMMON_INCLUDES-y)
+
+LIBPNG_CINCLUDES   += -I$(LIBPNG_EXTRACTED)
+
+################################################################################
+# Global flags
+################################################################################
+# Suppressed flags
+LIBPNG_SUPPRESS_FLAGS += -Wno-unused-parameter \
+	-Wno-unused-variable -Wno-unused-value -Wno-unused-function \
+	-Wno-missing-field-initializers -Wno-implicit-fallthrough
+LIBPNG_CFLAGS-y   += $(LIBPNG_SUPPRESS_FLAGS) \
+	-Wno-pointer-to-int-cast -Wno-int-to-pointer-cast
+
+LIBPNG_DEFINES    += -DHAVE_CONFIG_H
+LIBPNG_DEFINES    += -DHAVE_FEENABLEEXCEPT
+LIBPNG_CFLAGS-y   += $(LIBPNG_DEFINES)
+
+################################################################################
+# PNG src
+################################################################################
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/contrib/tools/pngfix.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/png.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngerror.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngget.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngmem.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngpread.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngread.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngrio.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngrtran.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngrutil.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngset.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngtrans.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngwio.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngwrite.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngwtran.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/pngwutil.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/mips/mips_init.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/mips/filter_msa_intrinsics.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/intel/intel_init.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/intel/filter_sse2_intrinsics.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/powerpc/powerpc_init.c
+LIBPNG_SRCS-y += $(LIBPNG_EXTRACTED)/powerpc/filter_vsx_intrinsics.c
+
+
+################################################################################
+# PNG prepare
+################################################################################
+# Run ./configure
+$(LIBPNG_EXTRACTED)/config.status: $(LIBPNG_BUILD)/.origin
+	$(call verbose_cmd,CONFIG,libpng: $(notdir $@), \
+		cd $(LIBPNG_EXTRACTED) && ./configure &&\
+		cp -f $(LIBPNG_EXTRACTED)/scripts/pnglibconf.h.prebuilt $(LIBPNG_EXTRACTED)/pnglibconf.h)
```

Comment:
This step needs to be removed.  The `pnglibconf.h` file needs to be explicitly created and placed within a to-level `include/` directory.  The directory and corresponding header file should be referenced via `LIBPNG_COMMON_INCLUDES`.  It's options need be checked and must correspond with what is compatible with Unikraft.